### PR TITLE
is0401: ensure Nodes correctly interpret registry TXT records

### PR DESCRIPTION
**Pull request by andrewbonney**
Another one that I think may require some feedback. I'm attempting to add a test to ensure that Nodes only register with registries which have suitable TXT records.

The trouble is that when a Node fails this test, lots of other tests fail because the Node has registered with the invalid first registry. As such it could make a Node look far less compliant with the spec than it really is. Potential resolutions include re-ordering the prereqs tests so that the invalid registry test happens after the others (however this will incur an additional 'DNS_SD_ADVERT_TIMEOUT' wait period). An alternative would be to add a further IS-04 Node test suite which covers some more time consuming tests. Any thoughts?

Notes:
* Untested with unicast so far
* Will require an exception for v1.0 as this didn't use the additional text records under test here

### CODE REVIEW

**Comment by garethsb-sony** [comment]
I haven't looked at the changes at all yet, but would one option be to disable (5xx) an invalid registry as soon as the Node registers with it, which should force the Node to move on to the valid ones?

**Comment by andrewbonney** [comment]
> I haven't looked at the changes at all yet, but would one option be to disable (5xx) an invalid registry as soon as the Node registers with it, which should force the Node to move on to the valid ones?

Good thought. I may actually be able to get the invalid one to return 500s immediately despite capturing what was POSTed.

File: IS0401Test.py [hunk 1]
```diff
@@ -82,40 +90,52 @@ def do_registry_basics_prereqs(self):
...
+
+            # Add advertisement with invalid version
+            info = self._registry_mdns_info(self.invalid_registry.get_data().port, priority, "v9.0")
+            registry_mdns.append(info)
+            # Add advertisement with invalid protocol
+            info = self._registry_mdns_info(self.invalid_registry.get_data().port, priority, None, "invalid")
+            registry_mdns.append(info)
+
+            # Add advertisement for primary and failover registries
             for registry in self.registries:
```
Comment (garethsb-sony) [inline IS0401Test.py]: Where has the construction of the `registry_mdns` info for the primary registry gone, now it's been separated out from the others by line 40?

Comment (garethsb-sony) [reply IS0401Test.py]: Oh no, my mistake, it's only the invalid registry that's been removed from the passed in `registries`.

**Comment by andrewbonney** [comment]
I just need to address the bulleted items before this gets merged, and an issue it's found in our implementation...

**Comment by garethsb-sony** [comment]
Oh yes, perhaps a bit eager with the 'approved' button, sorry.

**Comment by andrewbonney** [comment]
Now tested using mDNS and uDNS.

**Pull request merged**